src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -675,7 +675,7 @@ case object CREATE extends OpCode(0xf0, 3, 1, _.G_create) {
     } else {
 
       val (initCode, memory1) = state.memory.load(inOffset, inSize)
-      val (newAddress, world1) = state.world.newAddress(state.env.ownerAddr)
+      val (newAddress, world1) = state.world.increaseCreatorNonceAndCreateAddress(state.env.ownerAddr)
```

Comment:
Not always, this is not done when the tx is for contract creation (in which case the nonce is increased and the address created but in separate places). I agree that the name is misleading so I'll follow your name recommendation :+1: